torch/CMakeLists.txt

```diff
@@ -340,6 +340,12 @@ endif()
 
 
 add_library(torch_python SHARED ${TORCH_PYTHON_SRCS})
+if(DEFINED ENV{BUILD_CAFFE2})
```

Comment:
Did you add this? I'm surprised how this was not handled. Would the variable `BUILD_CAFFE2` not work without this change?

Reply:
`BUILD_CAFFE2` exists for a long time, but it was always consumed from main `CMakeLists.txt`. 
However, we need this to be propagated to `torch/CMakeLists.txt`. I have tried everything, including the standard `if(BUILD_CAFFE2)`, but for some reason `OPTIONS()` from root `CMakeLists.txt` are not propagated to `torch/CMakeLists.txt`. It kind of make sense because I couldnt find a `add_subdirectory(torch)` on the root dir, which enables such sharing.

Finally I mimic'ed this from other places in the pytorch build script.